### PR TITLE
Complete GeneralInfo(Sign Up)

### PR DESCRIPTION
**Pull request by MechanicKim**
Added 2 modules

react-native-image-picker(https://github.com/marcshilling/react-native-image-picker)
react-native-datepicker(https://github.com/xgfe/react-native-datepicker)

rnpm link does not work completely.
You might follow manual installation.

### CODE REVIEW

File: android/captures/com.yoda_2016.10.20_17.10.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
+Applications Memory Usage (kB):
```
Comment (ochanje210) [inline android/captures/com.yoda_2016.10.20_17.10.txt:1]: do you need this file to be in the repository?

File: src/components/Login.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ class Login extends Component {
     if (result === undefined) {
       Actions.login();
     } else {
-      Actions.main();
+      Actions.generalInfo();
```
Comment (ochanje210) [inline src/components/Login.js]: This is just general question but, is there a difference regarding performance between doing Actions.sceneName and doing Actions.push and pop?

File: src/components/SignUp/EduForm.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      </ScrollView>
+    );
+  }
+
+  toggleEdit() {
+    this.setState({
+      editMode: !this.state.editMode,
+    });
+  }
+
```
Comment (ochanje210) [inline src/components/SignUp/EduForm.js:171]: spaccccceeeeeeeeeee

**Comment by ochanje210** [comment]
@MechanicKim impressive work! and a huge update!
why don't you make PR more frequently in smaller amounts so that people can review and follow up easily

File: src/components/App.js [hunk 4]
```diff
@@ -78,4 +86,21 @@ class App extends Component {
   }
 }
 
+const styles = StyleSheet.create({
+  bar: {
+    backgroundColor: '#fafafa',
+    borderBottomColor: '#d6dada',
+  },
+
```
Comment (ace0625) [inline src/components/App.js]: no space here

File: src/components/SignUp/EditForm.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import {
+  StyleSheet,
```
Comment (ace0625) [inline src/components/SignUp/EditForm.js:4]: sort by alphabetical order

File: src/components/SignUp/EduForm.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
+import React, { Component } from 'react';
+import {
+  TouchableWithoutFeedback,
```
Comment (ace0625) [inline src/components/SignUp/EduForm.js]: sort by alphabetical order

File: src/components/SignUp/FileUploader.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      'Content-Type': 'application/json; charset=utf-8',
+    },
+    fields: null,
+    files: null,
+  };
+
+  upload(files, fields) {
+    this.option.files = files;
+    this.option.fields = {info: escape(JSON.stringify(fields))};
+    console.log('Send');
```
Comment (ace0625) [inline src/components/SignUp/FileUploader.js]: Delete log

File: src/components/SignUp/GeneralInfo.js [hunk 8]
```diff
@@ -1,29 +1,402 @@
 import React, { Component } from 'react';
 import {
   StyleSheet,
```
Comment (ace0625) [inline src/components/SignUp/GeneralInfo.js]: sort alphabetical order

File: src/components/SignUp/GeneralInfo.js [hunk 9]
```diff
@@ -1,29 +1,402 @@
...
+      workDataSource: workDS.cloneWithRows([]),
+      imageResource: null,
+    };
+  }
+
+  // After rendering, request user profile to server
+  componentDidMount() {
+    let onSuccess = (result) => this.onSuccess(result);
+    let onError = (error) => this.onError(error);
+    ServerUtil.initCallback(onSuccess, onError);
```
Comment (ace0625) [inline src/components/SignUp/GeneralInfo.js]: Shouldn't callback function be initialized in constructor?

File: src/components/SignUp/GeneralInfo.js [hunk 10]
```diff
@@ -1,29 +1,402 @@
...
+        marginTop: 64,
+      },
+      android: {
+        marginTop: 54,
+      },
+    }),
+    flex: 1,
+    flexDirection: 'column',
+  },
+
```
Comment (ace0625) [inline src/components/SignUp/GeneralInfo.js]: no empty lines

File: src/components/SignUp/Progress.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  View,
+} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+// This is a progress line.
+// This component receives 'level' and 'step' to define line length
+class Progress extends Component {
+
+  render() {
+    let deviceWidth = Dimensions.get('window').width;
```
Comment (ace0625) [inline src/components/SignUp/Progress.js]: devicewidth has already been defined on line 30

File: src/utils/ServerUtil.js [hunk 12]
```diff
@@ -169,6 +169,7 @@ class ServerUtil {
   }
 
   getException(error) {
+    console.log(error);
```
Comment (ace0625) [inline src/utils/ServerUtil.js]: delete

File: android/captures/com.yoda_2016.10.20_17.10.txt (hunk 1 above)
Comment (MechanicKim) [reply android/captures/com.yoda_2016.10.20_17.10.txt:1]: I don't know why this file was created.
I'll organize it. Thank you!

File: src/components/Login.js (hunk 2 above)
Comment (MechanicKim) [reply src/components/Login.js]: I think that there is no difference about performance.
Does anybody know about that?

File: android/captures/com.yoda_2016.10.20_17.10.txt (hunk 1 above)
Comment (ochanje210) [reply android/captures/com.yoda_2016.10.20_17.10.txt:1]: I guess u can simply add it to .gitignore

File: src/utils/UrlMeta.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 class UrlMeta {
 
   //Host and API
-  static host = 'http://ec2-52-78-121-221.ap-northeast-2.compute.amazonaws.com:8000/';
+  static host = 'http://192.168.0.53:8000/';
```
Comment (SarahhhC) [inline src/utils/UrlMeta.js]: Let's change this! This was for test.

File: src/utils/UrlMeta.js [hunk 14]
```diff
@@ -10,6 +10,7 @@ class UrlMeta {
   static API_ACTIVITY = 'match/activity/';
   static API_MENTOR_REQ = 'match/request/';
   static API_MENTOR_RESP = 'match/response/';
+  static API_USER_EDIT = 'users/edit/';
```
Comment (SarahhhC) [inline src/utils/UrlMeta.js]: url changed as `users/editGeneral`

Comment (MechanicKim) [reply src/utils/UrlMeta.js]: I got it! Thanks.

File: src/components/SignUp/MyPic.js [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+                 source={source} />
+        </TouchableWithoutFeedback>
+      </View>
+    );
+  }
+
+  showPicker() {
+    ImagePicker.showImagePicker(options, (response)  => {
+      if (response.didCancel) {
+        console.log('User cancelled image picker');
```
Comment (ace0625) [inline src/components/SignUp/MyPic.js]: Remove logs

File: src/components/Login.js (hunk 2 above)
Comment (ochanje210) [reply src/components/Login.js]: just letting you know, there is absolute difference where Actions.{name} is same as Action push and never pops which will keep scene stacks growing and result poor performance.

File: src/components/SignUp/WorkForm.js [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+  Text,
+  TextInput,
+  TouchableWithoutFeedback,
+  View,
+} from 'react-native';
+import DatePicker from 'react-native-datepicker';
+import styles from './Styles';
+
+class WorkForm extends Component {
+
```
Comment (ace0625) [inline src/components/SignUp/WorkForm.js]: Remove the whitespace

File: src/components/SignUp/FileUploader.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    fields: null,
+    files: null,
+  };
+
+  upload(files, fields) {
+    this.option.files = files;
+    this.option.fields = {info: escape(JSON.stringify(fields))};
+
+    // FileUpload.upload(this.option, (error, result) => {
```
Comment (kayla-kwon) [inline src/components/SignUp/FileUploader.js]: It's would be good to delete if this is not in use.

Comment (MechanicKim) [reply src/components/SignUp/FileUploader.js]: Oh! Good point!
Thanks!

File: ios/Bridgeme.xcodeproj/project.pbxproj [hunk 18]
```diff
@@ -38,8 +37,13 @@
 		CAD4F49CCA74414DB6CE0F54 /* Foundation.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 39817FD5C2A845CD8964A8D1 /* Foundation.ttf */; };
 		DAB407959AAC40BC9D08F885 /* Entypo.ttf in Resources */ = {isa = PBXBuildFile; fileRef = AFAEADFECB4B46DCAF4C4EAC /* Entypo.ttf */; };
 		DAED1D04D05544F98EBCFE9D /* libRCTFBSDK.a in Frameworks */ = {isa = PBXBuildFile; fileRef = C5DB8359243A4142B3E50AFF /* libRCTFBSDK.a */; };
+<<<<<<< HEAD:ios/Yoda.xcodeproj/project.pbxproj
+		C7863A15678044F6BAAF9F91 /* libBVLinearGradient.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 6B201487004F4267864CA47E /* libBVLinearGradient.a */; };
+		46B97FC1210140F0A9568E74 /* libRNImagePicker.a in Frameworks */ = {isa = PBXBuildFile; fileRef = E51CD6A690CC45538F3228D3 /* libRNImagePicker.a */; };
+=======
```
Comment (SarahhhC) [inline ios/Bridgeme.xcodeproj/project.pbxproj]: I think those `<<< HEAD === >>>` something are from merging conflict.
Please handle or clean them!

File: android/app/src/main/java/com/yoda/MainActivity.java [hunk 19]
```diff
@@ -4,7 +4,10 @@
 import android.os.Bundle;
 import com.facebook.react.ReactActivity;
 import com.BV.LinearGradient.LinearGradientPackage;
-import com.oblador.vectoricons.VectorIconsPackage;
+import com.imagepicker.ImagePickerPackage;
+import com.facebook.reactnative.androidsdk.FBSDKPackage;
+import com.BV.LinearGradient.LinearGradientPackage;
+import com.facebook.reactnative.androidsdk.FBSDKPackage;
```
Comment (SarahhhC) [inline android/app/src/main/java/com/yoda/MainActivity.java]: Why don't you sort them alphabetically~~?

File: android/app/src/main/java/com/yoda/MainActivity.java [hunk 20]
```diff
@@ -4,7 +4,10 @@
...
```
Comment (SarahhhC) [inline android/app/src/main/java/com/yoda/MainActivity.java]: Why don't you sort them alphabetically~~?

File: android/app/src/main/java/com/yoda/MainApplication.java [hunk 21]
```diff
@@ -6,10 +6,12 @@
 import com.facebook.FacebookSdk;
 import com.facebook.appevents.AppEventsLogger;
 import com.facebook.react.ReactApplication;
+import com.BV.LinearGradient.LinearGradientPackage;
+import com.imagepicker.ImagePickerPackage;
+import com.facebook.reactnative.androidsdk.FBSDKPackage;
```
Comment (SarahhhC) [inline android/app/src/main/java/com/yoda/MainApplication.java:8]: Why don't you sort them alphabetically~~?

File: src/components/App.js [hunk 22]
```diff
@@ -52,6 +53,8 @@ class App extends Component {
     let backAndroidHandler = () => {
       if (App.scene.sceneKey === 'evalPageMain' || App.scene.sceneKey == 'main') {
         return true;
+      } else if (App.scene.sceneKey === 'generalInfo') {
+        return true;
```
Comment (SarahhhC) [inline src/components/App.js:55]: I think if and else if returns same thing right? Why did you divide this??

File: src/components/SignUp/GeneralInfo.js [hunk 23]
```diff
@@ -1,29 +1,437 @@
...
+    }
+
+    return Input;
+  }
+
+  getDefaultEdu(edu, sectionID, rowID) {
+    let eduName = edu.school === undefined ? '' : edu.school.name;
+    let startYear = edu.startYear === undefined ? '' : edu.startYear.name;
+    let endYear = edu.year === undefined ? '' : edu.year.name;
+    if (edu.endYear !== undefined) endYear = edu.endYear.name;
```
Comment (SarahhhC) [inline src/components/SignUp/GeneralInfo.js:282]: Does this contain facebook and local login both?
Where did you handle facebook's 'year'?? 🤔

File: android/app/src/main/java/com/yoda/MainApplication.java (hunk 21 above)
Comment (MechanicKim) [reply android/app/src/main/java/com/yoda/MainApplication.java:8]: I wonder why it is important?

File: src/components/SignUp/GeneralInfo.js (hunk 23 above)
Comment (MechanicKim) [reply src/components/SignUp/GeneralInfo.js:282]: When I sign in first, there is no start and end fields. So, I think checking is nesseary.

File: android/app/src/main/java/com/yoda/MainApplication.java (hunk 21 above)
Comment (SarahhhC) [reply android/app/src/main/java/com/yoda/MainApplication.java:8]: If you think it's not necessary, then you can just let them.

**Comment by SarahhhC** [comment]
Good! Thanks @MechanicKim 👍 LGTM

**Comment by ochanje210** [comment]
good work!

**Pull request merged**